src/params.rs

```diff
@@ -207,3 +212,58 @@ impl FromStr for GuestMemorySize {
 		Ok(memory_size)
 	}
 }
+
+/// Configure the kernels environment variables.
+#[derive(Debug, Clone, PartialEq)]
+pub enum EnvVars {
+	/// Pass all env vars of the host to the kernel.
+	Host,
+	/// Pass a certain set of env vars to the kernel.
+	Set(HashMap<String, String>),
+}
+impl Default for EnvVars {
+	fn default() -> Self {
+		Self::Set(HashMap::new())
+	}
+}
+impl<S: AsRef<str> + std::fmt::Debug + PartialEq<S> + From<&'static str>> From<&[S]> for EnvVars {
+	fn from(v: &[S]) -> Self {
+		if v.contains(&S::from("host")) {
+			return Self::Host;
+		}
+		Self::Set(
+			v.iter()
+				.map(|s| {
+					let sep_pos = s
+						.as_ref()
+						.find('=')
+						.expect("Invalid environment variable format. Must be var=value");
+					let split = s.as_ref().split_at(sep_pos);
+					(String::from(split.0), String::from(&split.1[1..]))
```

Comment:
I think [`str::split_once`](https://doc.rust-lang.org/nightly/std/primitive.str.html#method.split_once) works better here.

This does not check for `-e K=v,A=b`, which might be confusing.

Reply:
Ah, I was actually searching for that function but must have overlooked it...